generate-readme.js

```diff
@@ -10,63 +10,64 @@ const log = require('./log')
 function toMarkdownList (a) {
   return a.map(d => `- ${d}`).join('\n')
 }
-function generate (dir, { pluginMeta, encapsulated, pluginFileName }, cb) {
+function generate (dir, { pluginMeta, encapsulated, pluginFileName }) {
   process.chdir(dir)
+  return new Promise((resolve, reject) => {
+    if (!existsSync(path.join(dir, 'package.json'))) {
+      execSync('npm init -y')
+      log('info', `generated package.json in ${dir}`)
+    }
 
-  if (!existsSync(path.join(dir, 'package.json'))) {
-    execSync('npm init -y')
-    log('info', `generated package.json in ${dir}`)
-  }
-
-  log('info', `reading package.json in ${dir}`)
-  let pkg = readFileSync('package.json')
-  try {
-    pkg = JSON.parse(pkg)
-  } catch (err) {
-    return cb(err)
-  }
+    log('info', `reading package.json in ${dir}`)
+    let pkg = readFileSync('package.json')
+    try {
+      pkg = JSON.parse(pkg)
+    } catch (err) {
+      reject(err)
```

Comment:
```suggestion
      return reject(err)
```

Reply:
I have a bit confused about why need to add `return` for `reject` in promise?

Reply:
reject() itself won't prevent the rest of the function from executing.

Reply:
@millette I have catch the error in [Line 123](https://github.com/fastify/fastify-cli/pull/184/files#diff-26bd854132ce1afd3ab68513953338edR123)

Reply:
Indeed, but within the function itself. For example https://github.com/xtx1130/fastify-cli/blob/promisify-generator/generate-readme.js#L62

The following line will run even `if reject(err)` is called.

```js
log('info', `README for plugin ${pkg.name} generated successfully`)
```